drivers/serial/uart_native_tty.c

```diff
@@ -311,8 +318,10 @@ static void native_tty_uart_irq_callback_set(const struct device *dev,
 
 static void native_tty_irq_init(const struct device *dev)
 {
+	struct native_tty_data *data = dev->data;
+
 	/* Create a thread which will wait for data - replacement for IRQ */
-	k_thread_create(&rx_thread, rx_stack, K_KERNEL_STACK_SIZEOF(rx_stack),
+	k_thread_create(&data->rx_thread, data->rx_stack, K_KERNEL_STACK_SIZEOF(data->rx_stack),
```

Comment:
`K_KERNEL_STACK_SIZEOF(data->rx_stack)` will give incorrect value since the macro doesn't work correctly with pointer arguments. It should be used only with arrays.

I think a better approach would be to define stack inside of the data structure using `K_KERNEL_STACK_MEMBER()`.

Reply:
Thanks for pointing this out!
I will fix this.

Reply:
@msalau I added `K_KERNEL_STACK_MEMBER()` as requested.

Reply:
Perfect!
Please also remove `DEFINE_RX_STACK()` and `ASSIGN_RX_STACK()`, since they are not useful anymore.

Reply:
And the stack is not a pointer anymore so please fix the comment too
`/* Pointer to the RX thread stack. */`

Reply:
Right, I forgot that.
Removed now.